tests/test_inventory_events_consumer.py

```diff
@@ -103,11 +108,54 @@ def test_host_update_event_no_cp(inventory_event_consumer, inventory_event_messa
         assert updated_system.display_name != updated_display_name
 
 
-def test_host_delete_event(inventory_event_consumer, db_setup):
-    msg = {"type": "delete", "insights_id": "677fb960-e164-48a4-929f-59e2d917b444",
+def test_host_delete_event(inventory_event_consumer, inventory_event_message, db_setup, mocker):
+    mocker.patch.object(
+        inventory_event_consumer,
+        'process_system_details',
+        side_effect=inventory_event_consumer.process_system_details,
+        autospec=True
+    )
```

Comment:
I think you don't need to mock this? in other tests as example `test_host_update_event_no_cp` we are having this call `inventory_event_consumer.process_system_details.call_count = 2` which needs the patching however in this test its not there?

Reply:
@upadhyeammit, that was the problem. In past, this test `test_host_delete_event` was useless in my opinion as without creating system it was trying to test deletion of system method. `rows_deleted.rowcount` was always zero. To fix this behavior I have added code that performs creation first and then test should verify whether deletion works when system exists. Please correct if I misunderstood your comment.